liblepton/src/i_vars.c

```diff
@@ -40,43 +37,111 @@ int   default_make_backup_files = TRUE;
 
 
 
-/*! \brief Initialize variables in TOPLEVEL object
- *  \par Function Description
- *  This function will initialize variables to default values.
+/* \brief Read a boolean configuration key.
  *
- *  \param [in] toplevel  The TOPLEVEL object to be updated.
+ * \par Function Description
+ * On success, set \a result to the value of the
+ * configuration key, otherwise set it to \a defval.
+ *
+ * \todo  Refactoring: this function was copied as is from schematic/src/i_vars.c.
+ *
+ * \param [in]       group   Configuration group name
+ * \param [in]       key     Configuration key name
+ * \param [in]       defval  Default value
+ * \param [in, out]  result  Result
+ *
+ * \return  TRUE if a specified config parameter was successfully read
+ */
+static gboolean
+cfg_read_bool (const gchar* group,
+               const gchar* key,
+               gboolean     defval,
+               gboolean*    result)
+{
+  gchar*     cwd = g_get_current_dir();
+  EdaConfig* cfg = eda_config_get_context_for_path (cwd);
+  g_free (cwd);
+
+  GError*  err = NULL;
+  gboolean val = eda_config_get_boolean (cfg, group, key, &err);
+
+  gboolean success = err == NULL;
+  g_clear_error (&err);
+
+  *result = success ? val : defval;
+  return success;
+}
+
+
```

Comment:
A thought aloud: we could move config processing, say, to `liblepton/scheme/geda/config.scm` the way like it is done, e.g., for `lepton-netlist`.

Reply:
What is supposed to be moved to `geda/config.scm`?
And I don't see any code related to lepton-netlist there.
Could you please elaborate on this?

Reply:
I meant something for `liblepton` like we have now for netlister in `netlist/scheme/netlist/config.scm`, that is, processing of config settings in Scheme.

Reply:
I thought about that, too.